static/schemas/2.0/skus.2.0.json

```diff
@@ -0,0 +1,248 @@
+                    "ipv4": {
+                        "type": "string",
+                        "format": "ipv4"
+                    },
+                    "ipv6": {
+                        "type": "string"
+                    },
+                    "is": {
+                        "type": "string"
+                    },
+                    "lowercase": {
+                        "type": "string"
+                    },
+                    "max": {
+                        "type": "string"
+                    },
+                    "maxLength": {
+                        "type": "string"
+                    },
+                    "min": {
+                        "type": "string"
+                    },
+                    "minLength": {
+                        "type": "string"
+                    },
+                    "notRegex": {
+                        "type": "string"
+                    },
+                    "notContains": {
+                        "type": "string"
+                    },
+                    "notIn": {
+                        "type": "string"
+                    },
+                    "notNull": {
+                        "type": "string"
+                    },
+                    "numeric": {
+                        "type": "string"
+                    },
+                    "required": {
+                        "type": "string",
+                        "format": "date"
+                    },
+                    "regex": {
+                        "type": "string"
+                    },
+                    "truthy": {
+                        "type": "string"
+                    },
+                    "uppercase": {
+                        "type": "string"
+                    },
+                    "url": {
+                        "type": "string",
+                        "format": "url"
+                    },
+                    "urlish": {
+                        "type": "string"
+                    },
+                    "uuid": {
+                        "type": "string"
+                    },
+                    "uuidv3": {
+                        "type": "string"
+                    },
+                    "uuidv4": {
+                        "type": "string"
+                    }
+                }
+            }
+        },
+        "SkusUpsert": {
+            "description": "A sku for RackHD",
+            "type": "object",
+            "required": [
+                "name"
+            ],
+            "properties": {
+                "name": {
+                    "type": "string"
+                },
+                "rules": {
+                    "$ref": "#/definitions/SkusRules"
+                },
+                "discoveryGraphName": {
+                    "type": "string"
+                },
+                "discoveryGraphOptions": {
+                    "type": "object",
+                    "properties": {
+                        "username": {
```

Comment:
Though it's not required, but you are trying to restrict their type.  I know usually the `username` `password` `hostname` is string, but it's free to customer to define the `hostname` as number in its customized discovery graph. this is not the scope for this general SKU schema.